build/configs/rtl8721csm/rtl8721csm_download.sh

```diff
@@ -275,19 +279,79 @@ download_all()
 		echo "Downloading ${parts[$partidx]} binary"
 		echo "=========================="
 
-		./amebad_image_tool $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
-		rm ${exe_name}
-
+		./amebad_image_tool 1 $TTYDEV 1 ${offsets[$partidx]} ${exe_name}
 	done
 	echo ""
 	echo "Download COMPLETE!"
 }
 
+erase()
+{
+	cd ${IMG_TOOL_PATH}
+	echo "Starting Erase..."
+	ota_addr=$(($FLASH_START_ADDR + $CONFIG_AMEBAD_FLASH_CAPACITY))
+	if [[ $1 == "erase_all" || $1 == "ERASE_ALL" ]];then
+		found_kernel=false
+		flash_ota=false
+		echo ""
+		echo "=========================="
+		echo "      Erasing All"
+		echo "=========================="
+		./amebad_image_tool 2 $TTYDEV 1 $FLASH_START_ADDR 0 $(($CONFIG_AMEBAD_FLASH_CAPACITY>>10))
+	else
+		for partidx in ${!parts[@]}; do
+			if [[ $1 == "erase_kernel" || $1 == "ERASE_KERNEL" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
+					ota_addr=0
+					found_kernel=false
+				else
+					continue
+				fi
+			elif [[ $1 == "erase_ota" || $1 == "ERASE_OTA" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ ${offsets[$partidx]} -lt ${ota_addr} ]];then
+					ota_addr=${offsets[$partidx]}
+					continue
+				else
+					flash_ota=false
+				fi
+			elif [[ $1 == "erase_userfs" || $1 == "ERASE_USERFS" ]];then
+				if [[ "${parts[$partidx]}" != "userfs" ]];then
+					continue
+				fi
+			fi
+			exe_name=$(get_executable_name ${parts[$partidx]})
+			[ "No Binary Match" = "${exe_name}" ] && continue
+
+			echo ""
+			echo "=========================="
+			echo "Erasing ${parts[$partidx]} binary"
+			echo "=========================="
+			./amebad_image_tool 2 $TTYDEV 1 ${offsets[$partidx]} ${exe_name} ${sizes[partidx]}
+		done
+	fi
+	echo ""
+	echo "Erase COMPLETE!"
+}
+
 case $1 in
 	all|ALL)
 		download_all
 		;;
+	erase_kernel|ERASE_KERNEL|erase_ota|ERASE_OTA|erase_userfs|ERASE_USERFS|erase_all|ERASE_ALL)
+		erase $1
+		;;
```

Comment:
If you change them as what I said above, you need to change here like below,
erase|ERASE)
 erase_specific_partition $1
 ;;

Reply:
Changing this one as well.